librz/core/cbin.c

```diff
@@ -368,6 +368,23 @@ RZ_API bool rz_core_bin_apply_all_info(RzCore *r, RzBinFile *binfile) {
 	return true;
 }
 
+RZ_API bool rz_core_file_reload(RzCore *r, const char *file, ut64 baseaddr) {
```

Comment:
This function should also be added to rz_core.h. Also, given it's a "reload" there should be no `file` argument (it is not even used) because it is assumed the current file is reloaded, i guess. Maybe, instead of passing the filename you could directly pass `RzBinFile` and move:
```
RzCoreFile *cf = rz_core_file_cur(r);
	if (!cf) {
		return false;
	}
	RzBinFile *obf = rz_bin_file_find_by_fd(r->bin, cf->fd);
	if (!obf) {
		return false;
	}
```
to the caller of this API. This is slightly better from an API point of view, in my opinion, because it allows you to reuse this API to also reload binaries that are not the current one.

Also, I think the new RzBinFile should be returned by this function instead of bool.